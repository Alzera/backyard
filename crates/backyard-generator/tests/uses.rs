use backyard_generator::generate;
use backyard_parser::parse_eval;

#[test]
fn basic() {
  let asts = parse_eval("use const A\\B;").unwrap();
  insta::assert_yaml_snapshot!(generate(&asts).unwrap());
}

#[test]
fn alias() {
  let asts = parse_eval("use const A\\B as A;").unwrap();
  insta::assert_yaml_snapshot!(generate(&asts).unwrap());
}

#[test]
fn multiple() {
  let asts = parse_eval("use A\\B as A, B\\C as B;").unwrap();
  insta::assert_yaml_snapshot!(generate(&asts).unwrap());
}

#[test]
fn items() {
  let asts = parse_eval(
    "use App\\Models\\{
  const User\\UserTesting as UserTestingA,
  User\\UserTestingB as UserTestingB,
  function UserTestingC
};"
  ).unwrap();
  insta::assert_yaml_snapshot!(generate(&asts).unwrap());
}
