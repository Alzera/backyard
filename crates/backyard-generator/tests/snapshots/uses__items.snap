---
source: crates/backyard-generator/tests/uses.rs
expression: generate(asts).unwrap()
snapshot_kind: text
---
"use App\\Models\\{\n  const User\\UserTesting as UserTestingA,\n  User\\UserTestingB as UserTestingB,\n  function UserTestingC\n};"
